core/src/components/checkbox/checkbox.tsx

```diff
@@ -176,6 +188,7 @@ export class Checkbox implements ComponentInterface {
           aria-checked={`${checked}`}
           disabled={disabled}
           id={inputId}
+          onChange={this.toggleChecked}
```

Comment:
Open to keeping this as an `onClick` listener. I did not see any noticeable differences, but using `onChange` may make the standalone form component change a bit easier for `ion-checkbox`.